.github/workflows/maven-release.yml

```diff
@@ -1,31 +1,62 @@
 name: Publish Release
-on:
-  push:
-    branches:
-      - main
+on: 
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'tag/version'
+        required: true        
+
+      action_tag:
+        description: 'Create tag? ("no" to skip)'
+        required: true
+        default: 'yes'
 jobs:
   publish:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+        with:
+          ref: main
+
       - name: Set up Java
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
-          java-version: 11
+          java-version: 17
```

Comment:
Add `distribution: 'zulu'` here, same as the [User Profile sample](https://github.com/adobe/aepsdk-userprofile-android/blob/dev-v3.0.0/.github/workflows/maven-release.yml#L22).  As the `setup-java` action supports different JDK profiles, I think it's best if each project uses the same one.
https://github.com/actions/setup-java?tab=readme-ov-file#supported-distributions